test/one/profiler/test/Runner.java

```diff
@@ -128,45 +127,48 @@ public static boolean enabled(Test test) {
-            String className = m.getDeclaringClass().getSimpleName();
-            String testName = className + '.' + m.getName();
+    private static TestResult run(RunnableTest rt) {
+        if (!enabled(rt.test()) || skipTests.contains(rt.className().toLowerCase()) || skipTests.contains(rt.method().getName().toLowerCase())) {
+            return TestResult.skip();
+        }
 
-            if (!enabled(test) || skipTests.contains(className.toLowerCase()) || skipTests.contains(m.getName().toLowerCase())) {
-                log.log(Level.FINE, "Skipped " + testName);
-                continue;
-            }
+        log.log(Level.INFO, "Running " + rt.testInfo() + "...");
+
+        String testLogDir = logDir.isEmpty() ? null : logDir + '/' + rt.testName();
+        try (TestProcess p = new TestProcess(rt.test(), currentOs, testLogDir)) {
+            Object holder = (rt.method().getModifiers() & Modifier.STATIC) == 0 ?
+                    rt.method().getDeclaringClass().getDeclaredConstructor().newInstance() : null;
+            rt.method().invoke(holder, p);
+        } catch (InvocationTargetException e) {
+            return TestResult.fail(e.getTargetException());
+        } catch (Throwable e) {
+            return TestResult.fail(e);
+        }
 
-            log.log(Level.INFO, "Running " + testName +
-                    (!test.args().isEmpty() ? " args: " + test.args() : "") +
-                    (test.inputs().length > 0 ? " inputs: [" + String.join(" ", test.inputs()) + "]" : "") +
-                    "...");
-
-            String testLogDir = logDir.isEmpty() ? null : logDir + '/' + testName;
-            try (TestProcess p = new TestProcess(test, currentOs, testLogDir)) {
-                Object holder = (m.getModifiers() & Modifier.STATIC) == 0 ?
-                        m.getDeclaringClass().getDeclaredConstructor().newInstance() : null;
-                m.invoke(holder, p);
-                log.info("OK");
-            } catch (InvocationTargetException e) {
-                log.log(Level.WARNING, testName + " failed", e.getTargetException());
-                passedAll = false;
-            } catch (Throwable e) {
-                log.log(Level.WARNING, testName + " failed", e);
-                passedAll = false;
+        return TestResult.pass();
+    }
+
+    private static List<RunnableTest> getRunnableTests(Class<?> cls) {
+        List<RunnableTest> rts = new ArrayList<>();
+        for (Method m : cls.getMethods()) {
+            for (Test t : m.getAnnotationsByType(Test.class)) {
+                rts.add(new RunnableTest(m, t));
             }
         }
-        return passedAll;
+        return rts;
     }
 
-    private static boolean run(Class<?> cls) {
-        boolean passedAll = true;
-        for (Method m : cls.getMethods()) {
-            passedAll &= run(m);
+    private static List<RunnableTest> getRunnableTests(String[] args) throws ClassNotFoundException {
```

Comment:
Tests that do not match current OS or current architecture are also considered runnable - is that intended?

Reply:
Yes, we collect all tests regardless of filtering here to count the total and report non-matching tests as skipped.

Reply:
This means I'll never see 100% tests passing... okay

Reply:
I have added percentage and skip reasons to the summary:

```
Total test duration: 24.4 ms
Results Summary:
PASS: 35
SKIP: 3
        ArchMismatch: 1
        OsMismatch: 2
FAIL: 0
TOTAL: 38 (100.00% PASS)
```